crates/async-sawtooth-sdk/src/zmq_client.rs

```diff
@@ -125,9 +120,9 @@ impl RequestResponseSawtoothChannel for ZmqRequestResponseSawtoothChannel {
                 .unwrap()
                 .send(message_type, &correlation_id, &bytes);
 
-            if let Err(e) = &future {
+            if let Err(e) = future {
                 error!(send_message=%correlation_id, error=%e);
-                //tx.send(Err(SawtoothCommunicationError::from(e.clone())));
+                tx.send(Err(SawtoothCommunicationError::from(e))).unwrap();
```

Comment:
The `unwrap()`'s a safe enough assumption? (Admittedly, already we do it again just below.)

Reply:
Yes, although we could introduce an ordering where the channel can no longer send, this wont happen with the current code base